bindings/http/http.go

```diff
@@ -99,6 +102,13 @@ func (h *HTTPSource) Invoke(ctx context.Context, req *bindings.InvokeRequest) (*
 				return nil, fmt.Errorf("invalid path: %s", path)
 			}
 		}
+
+		if treatNon200AsErrorStr, ok := req.Metadata["treatNon200AsError"]; ok {
+			treatNon200AsError = treatNon200AsErrorStr == "true"
```

Comment:
please use `IsTruthy` from `package metadata` for this

Reply:
Changed to use `IsTruthy` in places where string truthiness is evaluated